src/main/java/com/yourssu/roomescape/auth/RoleCheckInterceptor.java

```diff
@@ -0,0 +1,50 @@
+package com.yourssu.roomescape.auth;
+
+import com.yourssu.roomescape.infrastructure.CookieProvider;
+import com.yourssu.roomescape.member.Member;
+import com.yourssu.roomescape.member.MemberDao;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.Arrays;
+
+// todo: naming change
+
+@Component
+public class RoleCheckInterceptor implements HandlerInterceptor {
+    private final MemberDao memberDao;
+    private final LoginService loginService;
+
+    public RoleCheckInterceptor(LoginService loginService, MemberDao memberDao) {
+        this.loginService = loginService;
+        this.memberDao = memberDao;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        Cookie[] cookies = request.getCookies();
+        if (cookies == null) {
+            response.setStatus(400);
```

Comment:
```suggestion
            response.setStatus(HttpStatus.BAD_REQUEST.value());
```
HttpStatus 클래스를 사용하는건 어떨까요?